pkg/backend/filestate/storage/azure/storage.go

```diff
@@ -0,0 +1,36 @@
+package azure
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/pkg/errors"
+	"github.com/pulumi/pulumi/pkg/workspace"
+)
+
+// Login will handle getting the user's Azure Blob Storage
+// credentials and writing them into the current workspace.
+func Login(ctx context.Context, cloudURL string) error {
+	// If we have a saved access token, and it is valid, use it.
+	existingToken, err := workspace.GetAccessToken(cloudURL)
+	if err == nil && existingToken != "" {
+		//TODO: Validate existing token works
+		return nil
+	}
+
+	accessToken := os.Getenv(accessTokenEnvVar)
+
+	if accessToken != "" {
+		fmt.Printf("Using access token from %s\n", accessTokenEnvVar)
+	} else {
+		// TODO: Support other login modes
+		return errors.Errorf("Please login to your Azure backend using `pulumi login`")
```

Comment:
I'm not 100% sure this is an actionable error.  Presumably running `pulumi login` is going to fail, because it will just come back here and the envvar for the access token won't be set.  Should be be reading the token here interactively?

Reply:
Yeh sorry this was just a placeholder I left in before I implemented the prompt - I've added it in the latest commit.